src/libsystemd-network/dhcp-option.c

```diff
@@ -48,6 +49,35 @@ static int option_append(uint8_t options[], size_t size, size_t *offset,
                 *offset += 1;
                 break;
 
+        case SD_DHCP_OPTION_USER_CLASS: {
+                _cleanup_strv_free_ char **l = NULL;
+                unsigned i = 0;
+                char **s;
+
+                l = strv_split(optval, " ");
+
+                STRV_FOREACH(s, l) {
+                        i += strlen(*s) + 1;
+                }
+
+                if (size < *offset + i + 2)
+                        return -ENOBUFS;
+
+                options[*offset] = code;
```

Comment:
Please use `options[*(offset++)] = ...`.

Reply:
```
(gdb) r
Starting program: /home/sus/tt/dhcp-user-class/build/systemd-networkd 
Missing separate debuginfos, use: dnf debuginfo-install glibc-2.26-16.fc27.x86_64
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
host: Gained IPv6LL
guest: Gained IPv6LL
wlan0: Gained IPv6LL
Enumeration completed

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff74a5549 in __memmove_avx_unaligned_erms () from /lib64/libc.so.6
Missing separate debuginfos, use: dnf debuginfo-install cryptsetup-libs-1.7.5-3.fc27.x86_64 device-mapper-libs-1.02.144-1.fc27.x86_64 iptables-libs-1.6.1-4.fc27.x86_64 libacl-2.2.52-18.fc27.x86_64 libattr-2.4.47-21.fc27.x86_64 libblkid-2.30.2-1.fc27.x86_64 libcap-2.25-7.fc27.x86_64 libgcc-7.2.1-2.fc27.x86_64 libgcrypt-1.8.1-1.fc27.x86_64 libgpg-error-1.27-3.fc27.x86_64 libidn-1.33-4.fc27.x86_64 libpcap-1.8.1-6.fc27.x86_64 libseccomp-2.3.2-5.fc27.x86_64 libselinux-2.7-2.fc27.x86_64 libsepol-2.7-1.fc27.x86_64 libuuid-2.30.2-1.fc27.x86_64 lz4-libs-1.8.0-1.fc27.x86_64 pcre2-10.30-2.fc27.x86_64 systemd-libs-234-9.fc27.x86_64 xz-libs-5.2.3-4.fc27.x86_64
(gdb) bt
#0  0x00007ffff74a5549 in __memmove_avx_unaligned_erms () from /lib64/libc.so.6
#1  0x00005555555bebd6 in memcpy_safe (dst=0xbeefc53aa72ebdbd, src=0x5555558169b0, n=3) at ../src/basic/util.h:117
#2  0x00005555555bee2f in option_append (options=0x5555558361bc "5\001\001=\023\377F\314M\277", size=307, offset=0x7fffffffd298, code=77 'M', optlen=17, optval=0x555555820650)
    at ../src/libsystemd-network/dhcp-option.c:75
#3  0x00005555555bf044 in dhcp_option_append (message=0x5555558360cc, size=308, offset=0x7fffffffd288, overload=0 '\000', code=77 'M', optlen=17, optval=0x555555820650)
    at ../src/libsystemd-network/dhcp-option.c:112
#4  0x00005555555b59fe in client_send_discover (client=0x55555582bbf0) at ../src/libsystemd-network/sd-dhcp-client.c:772
#5  0x00005555555b640f in client_timeout_resend (s=0x55555583c7a0, usec=0, userdata=0x55555582bbf0) at ../src/libsystemd-network/sd-dhcp-client.c:1033
#6  0x00007ffff7ab8a6e in source_dispatch (s=0x55555583c7a0) at ../src/libsystemd/sd-event/sd-event.c:2278
#7  0x00007ffff7ab9d22 in sd_event_dispatch (e=0x5555558106d0) at ../src/libsystemd/sd-event/sd-event.c:2627
#8  0x00007ffff7aba140 in sd_event_run (e=0x5555558106d0, timeout=18446744073709551615) at ../src/libsystemd/sd-event/sd-event.c:2686
#9  0x00007ffff7aba27c in sd_event_loop (e=0x5555558106d0) at ../src/libsystemd/sd-event/sd-event.c:2706
#10 0x0000555555561c94 in main (argc=1, argv=0x7fffffffd5f8) at ../src/network/networkd.c:154
(gdb) f 2
#2  0x00005555555bee2f in option_append (options=0x5555558361bc "5\001\001=\023\377F\314M\277", size=307, offset=0x7fffffffd298, code=77 'M', optlen=17, optval=0x555555820650)
    at ../src/libsystemd-network/dhcp-option.c:75
75	                        memcpy_safe(&options[*offset + 1], *s, i);
```

Reply:
It just crashes with memory corruption. I guess I must be doing something wrong.

Reply:
Indeed. This should be `options[(*offset)++] = ...`.

Reply:
(In case it wasn't clear: it's better to do the increment in the same line in case of repeated assignment followed by increment.)

Reply:
I just kept the same pattern like we have in case default: and also I find it more readable